web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -212,15 +167,36 @@ public Promise<JsTable> getTable(Object key) {
         if (!JsArray.isArray(key)) {
             key = JsArray.of(key);
         }
-        List<Object> keyList = Js.<JsArray<Object>>uncheckedCast(key).asList();
-        // Every caller gets a fresh table instance, and when all are closed, the CTS will be released.
-        // See #populateLazyTable for how that is tracked.
-        final JsLazy<Promise<ClientTableState>> entry = tables.get(keyList);
-        if (entry == null) {
+        final List<Object> keyList = Js.<JsArray<Object>>uncheckedCast(key).asList();
+        if (!knownKeys.contains(keyList)) {
             // key doesn't even exist, just hand back a null table
             return Promise.resolve((JsTable) null);
         }
-        return entry.get().then(cts -> Promise.resolve(new JsTable(cts.getConnection(), cts)));
+        final ClientTableState entry = connection.newState((c, cts, metadata) -> {
+            // TODO deephaven-core#2529 parallelize this
+            connection.newTable(
+                    descriptor.getKeyColumnNamesList().asArray(new String[0]),
+                    keyColumnTypes.toArray(new String[0]),
+                    keyList.stream().map(item -> new Object[] {item}).toArray(Object[][]::new),
```

Comment:
Can you move this up to before the lambda, so we're certain the passed keyList won't have mutated by then? (I realize the original code was "wrong" in this same way.)